sources/packages/backend/libs/services/src/students-assessments/assessment-sequential-processing.service.ts

```diff
@@ -92,6 +105,124 @@ export class AssessmentSequentialProcessingService {
+      // There are no past applications.
+      return [];
+    }
+    const applicationNumbers = sequencedApplications.previous.map(
+      (application) => application.applicationNumber,
+    );
+    const totals = await this.studentAssessmentRepo
+      .createQueryBuilder("studentAssessment")
+      .select("disbursementValue.valueCode", "valueCode")
+      .addSelect("offering.offeringIntensity", "offeringIntensity")
+      .addSelect("SUM(disbursementValue.valueAmount)", "total")
+      .innerJoin("studentAssessment.application", "application")
+      .innerJoin(
+        "studentAssessment.disbursementSchedules",
+        "disbursementSchedule",
+      )
+      .innerJoin("disbursementSchedule.disbursementValues", "disbursementValue")
+      .innerJoin("studentAssessment.offering", "offering")
+      .where("application.applicationNumber IN (:...applicationNumbers)", {
+        applicationNumbers,
+      })
+      // Overwritten application can have awards associated with and they should not be considered.
+      .andWhere("application.applicationStatus != :overwrittenStatus", {
+        overwrittenStatus: ApplicationStatus.Overwritten,
+      })
+      // Check for assessment completed status to avoid retrieving any cancelation status.
+      .andWhere(
+        "studentAssessment.studentAssessmentStatus = :completedStudentAssessmentStatus",
+        {
+          completedStudentAssessmentStatus: StudentAssessmentStatus.Completed,
+        },
+      )
+      // Only consider disbursements in Pending, ReadyToSend, or Sent.
+      .andWhere(
+        "disbursementSchedule.disbursementScheduleStatus != :cancelledDisbursementScheduleStatus",
+        {
+          cancelledDisbursementScheduleStatus:
+            DisbursementScheduleStatus.Cancelled,
+        },
+      )
+      // Sequenced applications need at least one valid COE, which means that one can be cancelled
+      // and the other still valid. This ensures that only the valid one will be considered.
+      .andWhere("disbursementSchedule.coeStatus != :declinedCOEStatus", {
+        declinedCOEStatus: COEStatus.declined,
+      })
+      // Ensures that only grants will be returned since loans and not needed.
+      .andWhere("disbursementValue.valueType IN (:...grantsValueTypes)", {
+        grantsValueTypes: [
+          DisbursementValueType.CanadaGrant,
+          DisbursementValueType.BCGrant,
+        ],
+      })
+      .groupBy("offering.offeringIntensity")
+      .addGroupBy("disbursementValue.valueCode")
+      .getRawMany<{
+        offeringIntensity: OfferingIntensity;
+        valueCode: string;
+        total: string;
+      }>();
+    // Parses the values from DB ensuring that the total will be properly converted to a number.
+    // The valueAmount from awards are mapped to string by Typeorm Postgres driver.
+    return totals.map((total) => ({
+      offeringIntensity: total.offeringIntensity,
+      valueCode: total.valueCode,
+      total: +total.total,
```

Comment:
Is the getRawMany returning the total as string.

Reply:
Please see: https://github.com/bcgov/SIMS/pull/2817#discussion_r1490091399